core/domain/learner_group_services.py

```diff
@@ -0,0 +1,633 @@
+                    'parent_topic_name': topic.name,
+                    'thumbnail_filename': subtopic.thumbnail_filename,
+                    'thumbnail_bg_color': subtopic.thumbnail_bg_color,
+                    'subtopic_mastery': None
+                }
+                matching_subtopic_syllabus_item_dicts.append(
+                    syllabus_subtopic_dict
+                )
+
+    return matching_subtopic_syllabus_item_dicts
+
+
+def get_matching_story_syllabus_item_dicts(
+    topic: topic_domain.Topic,
+    group_story_ids: List[str],
+    keyword: Optional[str] = None
+) -> List[story_domain.LearnerGroupSyllabusStorySummaryDict]:
+    """Returns the matching story syllabus item dicts of the given topic
+    that can be added to the learner group syllabus.
+
+    Args:
+        topic: Topic. The topic whose stories are to be searched.
+        group_story_ids: list(str). The story ids of the learner group.
+        keyword: Optional[str]. The keyword to search the stories. It is
+            compared with the title of the story if passed in arguments.
+
+    Returns:
+        list(dict). The matching story syllabus item dicts of the given topic.
+    """
+    story_ids = [
+        story.story_id for story in
+        topic.canonical_story_references
+        if (
+            story.story_id not in group_story_ids and
+            story.story_is_published is True
+        )
+    ]
+
+    matching_stories = story_fetchers.get_story_summaries_by_ids(story_ids)
+    stories = story_fetchers.get_stories_by_ids(story_ids)
+
+    matching_story_syllabus_item_dicts: List[
+        story_domain.LearnerGroupSyllabusStorySummaryDict] = []
+
+    for ind, story in enumerate(matching_stories):
+        if keyword is None or keyword in story.title.lower():
+            summary_dict = story.to_dict()
+            syllabus_story_dict: story_domain.LearnerGroupSyllabusStorySummaryDict = { # pylint: disable=line-too-long
+                'id': summary_dict['id'],
+                'title': summary_dict['title'],
+                'description': summary_dict['description'],
+                'language_code': summary_dict['language_code'],
+                'version': summary_dict['version'],
+                'node_titles': summary_dict['node_titles'],
+                'thumbnail_filename': summary_dict['thumbnail_filename'],
+                'thumbnail_bg_color': summary_dict['thumbnail_bg_color'],
+                'url_fragment': summary_dict['url_fragment'],
+                'story_model_created_on':
+                    summary_dict['story_model_created_on'],
+                'story_model_last_updated':
+                    summary_dict['story_model_last_updated'],
+                'story_is_published': True,
+                'completed_node_titles': [],
+                'all_node_dicts': [
+                    node.to_dict() for node in
+                    stories[ind].story_contents.nodes # type: ignore[union-attr]
```

Comment:
Why is this ignore needed?

Reply:
stories[ind] can have none based on the type of data it has but it should not be since it's corresponding story summary exists, have modified it as well to remove it.